### ISSUE TITLE
ECQL for greater GeoServer compatibility

### ISSUE THREAD

**Issue by jodygarnett** [geostyler/geostyler-cql-parser#86]
Discussion topic: For ECQL the key improvements is to go from "<PropertyName> <op> <Expression>" to "<Expression> <Op> <Expression>".

References:

* [CQL](https://github.com/geotools/geotools/blob/master/modules/library/cql/CQL.md)
* [ECQL](https://github.com/geotools/geotools/blob/master/modules/library/cql/ECQL.md)

**Comment by hwbllmnn** [comment]
I agree that this would be very nice to have. Unfortunately I'd guess that it's not easy to implement with the current state of the parser (it was originally just ported from OpenLayers 2). The best way forward would probably be a (partial) rewrite of the parser.

**Comment by jodygarnett** [comment]
I wonder if openlayers has any new code we can borrow.

As for ease of implementation, the code to parse an expression must already be present, it may be a manageable change to allow it to parse both the left and right hand side of an operation.

That said additional WFS 2.0 operations and filters have been defined (since any openlayers 2 parser was written), for example "between" to support temporal queries.
